### ISSUE TITLE
Does not support username:api token in the url

### ISSUE THREAD

**Issue by BobNowadly** [sep/jenkins-hue-monitor#3]
Please add an argument where I can pass in a [username]:[API Token for jenkins]

**Comment by mike-rogers** [comment]
Right-o. I created a new branch called `add_credentials` to support this feature. Kindly point your build at that branch and see if things work for you. You'll want to add `-c username:token` to your build's shell callout.

I don't expect it to work the first time. The unit tests pass but I don't have a setup on which I can test this all the way through. Feel free to grab me when you start testing this so we can shorten the feedback loop.

**Comment by BobNowadly** [comment]
Hey my Hue light build did this:

[EnvInject] - Loading node environment variables.
Building remotely on jenxins<http://jenkins.net.sep.com/computer/jenxins> (linux) in workspace /home/jenkins/jenkins-home/workspace/Hue status for eDMEplus
> /usr/bin/git rev-parse --is-inside-work-tree # timeout=10
Fetching changes from the remote Git repository
> /usr/bin/git config remote.origin.url https://github.com/sep/jenkins-hue-monitor.git # timeout=10
Fetching upstream changes from https://github.com/sep/jenkins-hue-monitor.git
> /usr/bin/git --version # timeout=10
> /usr/bin/git fetch --tags --progress https://github.com/sep/jenkins-hue-monitor.git +refs/heads/*:refs/remotes/origin/*
> /usr/bin/git rev-parse origin/add_credentials^{commit} # timeout=10
Checking out Revision 1b341413b34595c3ff8a5bd0154a5ef7d4b8035d (origin/add_credentials)
> /usr/bin/git config core.sparsecheckout # timeout=10
> /usr/bin/git checkout -f 1b341413b34595c3ff8a5bd0154a5ef7d4b8035d
> /usr/bin/git rev-list 1b341413b34595c3ff8a5bd0154a5ef7d4b8035d # timeout=10
[Hue status for eDMEplus] $ /bin/bash /tmp/hudson215506130924053894.sh
Fetching gem metadata from https://rubygems.org/.........
Resolving dependencies...
You have requested:
  trollop >= 0

The bundle currently has trollop locked at 2.0.
Try running `bundle update trollop`
Build step 'Execute shell' marked build as failure
Finished: FAILURE

**Comment by mike-rogers** [comment]
You could try clicking the checkbox in the build configuration that wipes out your workspace each build. If that doesn't work, come get me and we can bang on this in realtime.

**Comment by BobNowadly** [comment]
I tried that I will give it another shot though

**Comment by mike-rogers** [comment]
Give the `add_credentials` branch a shot again when you get a chance. You'll have to reset your workspace as I've been force-pushing like a monster.

**Comment by BobNowadly** [comment]
I have a successful Hue light build

**Comment by paolodm** [comment]
Sounds like this is working, no? Would you mind merging this into master?

**Comment by mike-rogers** [comment]
@paolodm I've been seeing error messages in different versions of Ruby, especially when looking at HTTPS-based Jenkins installations. I'm going to hold off on merging these changes until l have a change to resolve those errors. Until then you should be able to use the branch from within the Git plugin in Jenkins.
